labs/a3.md

```diff
@@ -0,0 +1,520 @@
+installing out system onto them!
+
+> Much of this will be straight from the ArchWiki installation guide,
+> starting at
+> [https://wiki.archlinux.org/index.php/Installation_guide#Installation](https://wiki.archlinux.org/index.php/Installation_guide#Installation).
+> It's a good idea to follow along there as well to see what
+> modifications we are making.
+
+### mirrors
+
+We need to tell Arch which software mirror to use when downloading
+packages. Go ahead and just put the OCF mirror at the top! Edit
+`/etc/pacman.d/mirrorlist` using `vim`/`emacs`/`nano` and put the
+following line at the top:
+
+```
+Server = https://mirrors.ocf.berkeley.edu/archlinux/$repo/os/$arch
+```
+
+(We're usually lower down in the file but putting us at the top will
+make things a bit faster to install.)
+
+### continue according to ArchWiki!
+
+Follow the arch wiki exactly from
+[https://wiki.archlinux.org/index.php/Installation_guide#Install_the_base_packages](https://wiki.archlinux.org/index.php/Installation_guide#Install_the_base_packages)
+down until you get to the "Initramfs" section. If you get stuck, check
+out the following hints:
+
+- `vim` is not installed in the chroot by default. You can do
+  `pacman -S vim` to install it, or just use plain old `vi`.
+- Our timezone is `America/Los_Angeles`.
+- The locale you want to use is `en_US.UTF-8`.
+- You shouldn't need to change the keyboard layout.
+- You can pick whatever you want for your hostname. OCF machines are
+  all named after "disasters", so it can be fun to come up with a
+  creative name that fit that theme.
+
+### initramfs
+
+The initramfs is the minimal stuff that's loaded into RAM that helps
+Linux get started when you first boot. On Arch, there's a system to
+generate this initramfs. You need to tell it to support disk
+encryption.
+
+Based on the [ArchWiki's
+instructions](https://wiki.archlinux.org/index.php/Dm-crypt/Encrypting_an_entire_system#Configuring_mkinitcpio),
+we need to edit `/etc/mkinitcpio.conf`. You should see a line like
+
+```
+HOOKS=(base udev autodetect ...)
+```
+
+We need to add the `keymap` and `encrypt` hooks, and move the
+`keyboard` hook earlier in the line. When you're done, it should look
+like
+
+```
+HOOKS=(base udev autodetect keyboard keymap modconf block encrypt filesystems fsck)
+```
+
+The _order_ of these hooks matters! Be careful.
+
+Run `mkinitcpio -p linux` to generate the new initramfs based on the
+new config file.
+
+### root password
+
+Use `passwd` to set the root password to `itoolovelinux`.
+
+> As mentioned for the disk encryption passphrase, don't use a
+> password like this in real life! Choose a strong root password using
+> a method like the one from earlier. User programs shouldn't be able
+> to compromise this password, because it would allow a single program
+> (which should generally be untrusted) to access everything on your
+> computer.
+
+### boot loader
+
+Historically, GRUB has been the only reasonable choice of boot
```

Comment:
what is a boot loader

Reply:
this is sort of the same thing as partitions: it might be nice to explain it here but it's also covered in depth in lecture

Reply:
:F: me